File: test_ws.py
import asyncio
import websockets
import json
import sys

async def test_ws():
    uri = "ws://127.0.0.1:8082/app/uni-chat-key?protocol=7&client=js&version=8.4.0-rc2&flash=false"
    try:
        async with websockets.connect(uri) as websocket:
            print("Connected!")
            greeting = await asyncio.wait_for(websocket.recv(), timeout=5.0)
            print(f"Received: {greeting}")
            
            sub_msg = json.dumps({
                "event": "pusher:subscribe",
                "data": {"channel": "chat.room.1"}
            })
            await websocket.send(sub_msg)
            print("Sent subscribe")
            
            res = await asyncio.wait_for(websocket.recv(), timeout=5.0)
            print(f"Received: {res}")
    except Exception as e:
        print(f"Error: {e}")

asyncio.run(test_ws())

File: test_scripts/test_ws.js
const WebSocket = require('ws');
const ws = new WebSocket('ws://127.0.0.1:8082/app/uni-chat-key?protocol=7&client=js&version=8.4.0-rc2&flash=false');

ws.on('open', function open() {
  console.log('Connected');
});

ws.on('message', function incoming(data) {
  console.log('Received: %s', data);
  const msg = JSON.parse(data);
  if (msg.event === 'pusher:connection_established') {
    const subscribeMsg = JSON.stringify({
      event: 'pusher:subscribe',
      data: { channel: 'chat.room.1' }
    });
    ws.send(subscribeMsg);
    console.log('Sent subscribe');
  }
});
